balancer.go

```diff
@@ -30,6 +30,10 @@ type (
 
 	Nodes []*Node
 
+	// StrategyFunc represent a factory that return a new Strategy
+	// the main role for this func is for users extend new strategies as needed
+	StrategyFunc func(nodes Nodes) (Strategy, error)
```

Comment:
suggestion: `StrategyFactory` provides more context to what this func should do.